satpy/multiscene.py

```diff
@@ -46,18 +46,26 @@
 log = logging.getLogger(__name__)
 
 
-def stack(datasets, weights=None, combine_times=True):
-    """Overlay a series of datasets together.
+def stack(datasets, weights=None, combine_times=True, blend_type='select_with_weights'):
+    """Combine a series of datasets in different ways.
 
     By default, datasets are stacked on top of each other, so the last one applied is
-    on top. If a sequence of weights arrays are provided the datasets will
-    be combined according to those weights. The result will be a composite
-    dataset where the data in each pixel is coming from the dataset having the
-    highest weight.
+    on top. If a sequence of weights (with equal shape) is provided, the datasets will
+    be combined according to those weights. Datasets can be integers like 'ct', 'cma',
+    or radiances single channel or RGB composites. In the later case weights is applied
+    to each 'R', 'G', 'B' coordinate in the same way. The result will be a composite
+    dataset where each pixel is constructed in a way depending on variable blend_type.
 
     """
-    if weights:
-        return _stack_weighted(datasets, weights, combine_times)
+    bands = datasets[0].dims[0] == 'bands'
```

Comment:
@djhoese  regarding your question above "**_This is essentially a weighted average, right?_** " the answer is YES. But for blend_type = "blend_with_weights" I also use my special "AzB" type weights that go to zero towards the boundaries of available data. Otherwise we would still see e.g. the ends of the scan lines for multi pass LEO images. In other words for blend_type = "blend_with_weights" a setting like (above) weights = (180. - satellite_zenith_angle") makes no sense.